services/libs/data-access-layer/src/collections/index.ts

```diff
@@ -326,61 +326,53 @@ export async function findBySlug(qx: QueryExecutor, slug: string) {
   return collections
 }
 
-export async function insertSegmentRepositories(
+export async function upsertSegmentRepositories(
   qx: QueryExecutor,
   {
     insightsProjectId,
     repositories,
     segmentId,
   }: {
-    insightsProjectId?: string
+    insightsProjectId: string
     repositories: string[]
     segmentId: string
   },
 ) {
   if (repositories.length === 0) {
-    return null
+    return
   }
 
-  const data = repositories.map((repo) => ({
-    insightsProjectId,
-    repository: repo,
-    segmentId,
-  }))
-
   return qx.result(
-    prepareBulkInsert(
-      'segmentRepositories',
-      ['insightsProjectId', 'repository', 'segmentId'],
-      data,
-      '("repository", "insightsProjectId") DO NOTHING',
-    ),
+    `
+    WITH "input" AS (
+      SELECT DISTINCT unnest(ARRAY[$(repositories:csv)]::text[]) AS "repository"
+    )
+    INSERT INTO "segmentRepositories" ("repository", "segmentId", "insightsProjectId")
+    SELECT "repository", $(segmentId), $(insightsProjectId)
+    FROM "input"
+    ON CONFLICT ("repository", "insightsProjectId")
```

Comment:
Should it be
ON CONFLICT ("repository") instead of ON CONFLICT ("repository", "insightsProjectId") .
Imagine a repository is already mapped in the table and you want to update it, the insightsProjectId would always be different, we only want to check the conflict on the repository side. So if we are trying to update the same repository, then both insightsProjectId and segmentId would be udpated to whatever we give it as input